lightning/mydump/region.go

```diff
@@ -171,6 +188,71 @@ func MakeTableRegions(
 		dataFileSizes = append(dataFileSizes, float64(dataFileSize))
 	}
 
-	AllocateEngineIDs(filesRegions, dataFileSizes, float64(batchSize), batchImportRatio, float64(tableConcurrency))
+	AllocateEngineIDs(filesRegions, dataFileSizes, float64(cfg.Mydumper.BatchSize), cfg.Mydumper.BatchImportRatio, float64(cfg.App.TableConcurrency))
 	return filesRegions, nil
 }
+
+// SplitLargeFile splits a large csv file into multiple regions, the size of
+// each regions is specified by `config.MaxRegionSize`.
+// Note: We split the file coarsely, thus the format of csv file is needed to be
+// strict.
+// e.g.
+// - CSV file with header is invalid
+// - a complete tuple split into multiple lines is invalid
+func SplitLargeFile(
+	meta *MDTableMeta,
+	cfg *config.Config,
+	dataFilePath string,
+	dataFileSize int64,
+	divisor int64,
+	prevRowIdxMax int64,
+	ioWorker *worker.Pool,
+) (prevRowIdMax int64, regions []*TableRegion, dataFileSizes []float64, err error) {
+	maxRegionSize := cfg.Mydumper.MaxRegionSize
+	dataFileSizes = make([]float64, 0, dataFileSize/maxRegionSize)
+	startOffset, endOffset := int64(0), maxRegionSize
+	for {
+		curRowsCnt := (endOffset - startOffset) / divisor
+		rowIDMax := prevRowIdxMax + curRowsCnt
+		if endOffset != dataFileSize {
+			reader, err := os.Open(dataFilePath)
```

Comment:
is os.Open() once enough here?

Reply:
No, parser.Close will close this fd. So we need to Open it again.